askama_derive/src/generator.rs

```diff
@@ -73,9 +93,84 @@ fn build_template(ast: &syn::DeriveInput) -> Result<String, CompileError> {
     if input.print == Print::Code || input.print == Print::All {
         eprintln!("{code}");
     }
+
+    #[cfg(feature = "cache")]
+    cache_template(&digest, &code)?;
+
     Ok(code)
 }
 
+#[cfg(feature = "cache")]
+fn get_or_create_cache_dir() -> Result<PathBuf, CompileError> {
+    use std::{env, fs};
+
+    let dir = if let Ok(dir) = env::var("ASKAMA_CACHE_DIR") {
+        PathBuf::from(dir)
+    } else {
+        let root = if let Ok(dir) = env::var("CARGO_TARGET_DIR") {
+            PathBuf::from(dir)
+        } else {
+            PathBuf::from("./target")
```

Comment:
Does this do the right thing in the face of workspaces? (That is, store `askama` in `<workspace>/target` rather than `<workspace>/<crate>/target`?)